tests/scripts/analyze_outcomes.py

```diff
@@ -51,6 +53,25 @@ def hits(self):
         """
         return len(self.successes) + len(self.failures)
 
+def execute_reference_driver_tests(ref_component, driver_component, outcome_file):
+    """Run the tests that will fullfill the outcome file used for the following
+    coverage analysis"""
+    # If the outcome file already exists, we assume that the user wants to
+    # perform the comparison analysis again without repeating the tests.
+    if os.path.exists(outcome_file):
+        Results.log("Outcome file (" + outcome_file + ") already exists. " + \
+                    "Tests will be skipped.")
+        return
+
+    shell_command = "tests/scripts/all.sh --outcome-file " + outcome_file + \
+                    " " + ref_component + " " + driver_component
+    print("Running: " + shell_command)
+    ret_val = subprocess.run(shell_command.split(), check=False).returncode
+
+    if ret_val != 0:
+        Results.log("Error: failed to run reference/driver components")
```

Comment:
Shouldn't this be Results.error? Also, it's not really a failure to run, but the fact that it returned an error (because, for example, one test suite didn't pass).

Reply:
Yes, even though the `error` function is not really used on this script, it seems to me that its goal was not stop the execution of the whole script, but to track an error and move on with testing (indeed it has an internal counter for that).
However, in this case, what I really wanted was just to print something on the screen and stop everything.
Indeed also the `subprocess` function is called with `check=False` which allows to reach the `Results.log` line instead of printing an ugly stack trace in case of error.